pkg/runtime/wfengine/workflowstate.go

```diff
@@ -203,47 +235,64 @@ func LoadWorkflowState(ctx context.Context, actorRuntime actors.Actors, actorID
 	if err = json.Unmarshal(res.Data, &metadata); err != nil {
 		return nil, fmt.Errorf("failed to unmarshal workflow metadata: %w", err)
 	}
+
+	// Load inbox, history, and custom status using a bulk request
 	state := NewWorkflowState(config)
 	state.Generation = metadata.Generation
-	// CONSIDER: Do some of these loads in parallel
+	state.Inbox = make([]*backend.HistoryEvent, metadata.InboxLength)
+	state.History = make([]*backend.HistoryEvent, metadata.HistoryLength)
+
+	bulkReq := &actors.GetBulkStateRequest{
+		ActorType: config.workflowActorType,
+		ActorID:   actorID,
+		Keys:      make([]string, metadata.InboxLength+metadata.HistoryLength+1),
```

Comment:
Nit: I assume the +1 is for custom status? If so, can we add a little comment stating this?